components/net/blob_loader.rs

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use hyper::header::{DispositionType, ContentDisposition, DispositionParam};
+use hyper::header::{Headers, ContentType, ContentLength, Charset};
+use hyper::http::RawStatus;
+use mime::{Mime, Attr};
+use mime_classifier::MIMEClassifier;
+use net_traits::ProgressMsg::Done;
+use net_traits::blob_url_store::{parse_blob_url, BlobURLStoreEntry, BlobURLStoreError};
+use net_traits::response::HttpsState;
+use net_traits::{LoadConsumer, LoadData, Metadata, NetworkError};
+use filemanager_thread::BlobURLStore;
+use resource_thread::{send_error, start_sending_sniffed_opt};
+use std::str;
+use std::sync::{Arc, RwLock};
+
+
+// TODO: Check on GET
+// https://w3c.github.io/FileAPI/#requestResponseModel
+
+pub fn load(load_data: LoadData, consumer: LoadConsumer,
+            blob_url_store: Arc<RwLock<BlobURLStore>>,
+            classifier: Arc<MIMEClassifier>) { // XXX: Move it into net process later
+
```

Comment:
@Manishearth Regarding `factory`, it is totally gone here. And it is just a function that will be called with the extended `FileManager`'s attributes.